### PR TITLE
test(ecocredit): add update project admin integration test

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- adds an integration test for updating a project admin

ref: #1047 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/client/testsuite/tx.go [hunk 1]
```diff
@@ -1426,6 +1427,84 @@ func (s *IntegrationTestSuite) TestTxBuyDirect() {
...
+	unauthorizedAccount, _, err := val.ClientCtx.Keyring.NewMnemonic("unauthorized", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+	unauthAddr := unauthorizedAccount.GetAddress()
+	s.fundAccount(clientCtx, val.Address, unauthAddr, sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 50)})
+
+	testCases := []struct {
+		name          string
+		args          []string
+		errMsg        string
+		errLog        string
```
Comment (technicallyty) [inline x/ecocredit/client/testsuite/tx.go:1450]: had to add 2 separate err strings here, the msg is caught by the CLI itself, Log is caught via the server impl so the expected error is returned in the log of `sdk.TxResponse`

**Comment by technicallyty** [comment]
not really sure why all the docs are failing.. 🤔

File: x/ecocredit/client/testsuite/tx.go [hunk 2]
```diff
@@ -1426,6 +1427,84 @@ func (s *IntegrationTestSuite) TestTxBuyDirect() {
 	}
 }
 
+func (s *IntegrationTestSuite) TestUpdateProjectAdmin() {
+	val := s.network.Validators[0]
+	updateTo := s.network.Validators[1].Address.String()
```
Comment (ryanchristo) [inline x/ecocredit/client/testsuite/tx.go:1432]: Might be nice to name these `admin` and `newAdmin` rather than `val` and `updateTo`.

**Review by ryanchristo — APPROVED** [review]
Looks ok to me. Seems like this would be more valuable within the server integration tests since we are planning to keep those and remove client integration tests in the future. We also cover this particular scenario in our unit tests but it doesn't hurt to have here now. Pre-approving.

**Comment by technicallyty** [comment]
> Looks ok to me. Seems like this would be more valuable within the server integration tests since we are planning to keep those and remove client integration tests in the future. We also cover this particular scenario in our unit tests but it doesn't hurt to have here now. Pre-approving.

should the issue be updated then? #1047 only mentions implementing CLI tests for these functions

File: x/ecocredit/client/testsuite/tx.go [hunk 3]
```diff
@@ -1426,6 +1427,84 @@ func (s *IntegrationTestSuite) TestTxBuyDirect() {
...
+	clientCtx := val.ClientCtx
+	_, projectId, _ := s.createClassProjectBatch(clientCtx, val.Address.String())
```
[the remark this answers is not in the document]
Comment (technicallyty) [reply x/ecocredit/client/testsuite/tx.go]: made a helper function in https://github.com/regen-network/regen-ledger/pull/1102/commits/d10c329bd8181e2f2129989c49b80e8d7f1673ee for just class + project

File: x/ecocredit/client/testsuite/tx.go [hunk 4]
```diff
@@ -1426,6 +1427,84 @@ func (s *IntegrationTestSuite) TestTxBuyDirect() {
...
+	cmd := coreclient.TxUpdateProjectAdminCmd()
+	makeArgs := func(projectId, newAdminAddr, from string) []string {
+		args := make([]string, 0, 6)
+		args = append(args, projectId, newAdminAddr, makeFlagFrom(from))
+		return append(args, s.commonTxFlags()...)
+	}
+	unauthorizedAccount, _, err := val.ClientCtx.Keyring.NewMnemonic("unauthorized", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+	unauthAddr := unauthorizedAccount.GetAddress()
```
[the remark this answers is not in the document]
Comment (technicallyty) [reply x/ecocredit/client/testsuite/tx.go]: fixed in https://github.com/regen-network/regen-ledger/pull/1102/commits/d10c329bd8181e2f2129989c49b80e8d7f1673ee

File: x/ecocredit/client/testsuite/tx.go (hunk 2 above)
Comment (technicallyty) [reply x/ecocredit/client/testsuite/tx.go]: renamed in https://github.com/regen-network/regen-ledger/pull/1102/commits/d10c329bd8181e2f2129989c49b80e8d7f1673ee

**Comment by ryanchristo** [comment]
> should the issue be updated then? https://github.com/regen-network/regen-ledger/issues/1047 only mentions implementing CLI tests for these functions

No need to update the issue. The issue is specific to CLI tests and was marked as a nice to have and we already have an issue for auditing server implementation tests https://github.com/regen-network/regen-ledger/issues/1042.

I think this is ok given its already implemented but it would have been nice to prioritize the server integration tests over the client integration tests given the ORM query service will make these obsolete in the future.

**Comment by ryanchristo** [comment]
> not really sure why all the docs are failing.. thinking

Looks like something to do with yarn as a dependency. I'll look into it but no need to resolve here.

**Comment by ryanchristo** [comment]
Just needed to update the netlify configuration to use Go 1.18. All good now.

**Pull request merged**